daemon/cmd/fqdn.go

```diff
@@ -400,6 +399,12 @@ func (d *Daemon) notifyOnDNSMsg(lookupTime time.Time, ep *endpoint.Endpoint, epI
 
 	stat.ProcessingTime.End(true)
 
+	serverAddrPort, err := netip.ParseAddrPort(serverAddr)
+	if err != nil {
+		log.WithError(err).Error("cannot extract destination IP/port from DNS request")
+	}
+	ep.UpdateProxyStatistics("fqdn", strings.ToUpper(protocol), serverAddrPort.Port(), proxy.DefaultDNSProxy.GetBindPort(), false, !msg.Response, verdict)
```

Comment:
I think we can avoid parsing address/port (again) here by passing a `netip.AddrPort` to `(*Daemon).notifyOnDNSMsg`. The only caller already has a `netip.AddrPort` which it converts to string before calling `notifyOnDNSMsg`.

I'll send a follow-up PR once this one is merged.

Reply:
hah, good observation! This whole flow is due for a refactor. No reason we should be holding on to the packet after policy updates are done; events and bookkeeping can wait.

Reply:
https://github.com/cilium/cilium/pull/37558